### PR TITLE
[performance] only `logLibraryVersions` in debug mode

### PR DESCRIPTION
**Pull request by GavinJoyce**
screen shot 2017-01-12 at 16 48 14

TODO:

 * [x] verify that the ember inspector continues to work: https://github.com/emberjs/ember-inspector/blob/679643037032fab3e4388a8dcb8cd0f20d1cc660/shared/in-page-script.js#L15

### CODE REVIEW

**Comment by rwjblue** [comment]
We should also remove the invocation [here](https://github.com/emberjs/ember.js/blob/d26c3fbbf05cf8ba372299df051b28e2fced25f7/packages/ember-application/lib/system/application.js#L342) so the whole function gets DCE by uglify.

File: packages/ember-application/lib/system/application.js [hunk 1]
```diff
@@ -339,7 +340,7 @@ const Application = Engine.extend({
     }
 
     registerLibraries();
-    logLibraryVersions();
+    runInDebug(() => logLibraryVersions());
```
Comment (locks) [inline packages/ember-application/lib/system/application.js]: do we need this with the no-opping body?

**Comment by GavinJoyce** [comment]
See Robert's comment above

**Review by locks — APPROVED** [review]

**Review by rwjblue — APPROVED** [review]

**Comment by rwjblue** [comment]
LGTM, will land once CI is finished.

**Comment by GavinJoyce** [comment]
🚢

**Pull request merged**